package/MDAnalysis/coordinates/GRO.py

```diff
@@ -269,19 +275,37 @@ def write(self, selection, frame=None):
            resName and atomName are truncated to a maximum of 5 characters
         """
         # write() method that complies with the Trajectory API
-        u = selection.universe
-        if frame is not None:
-            u.trajectory[frame]  # advance to frame
-        else:
-            frame = u.trajectory.ts.frame
 
-        # make sure to use atoms (Issue 46)
-        atoms = selection.atoms
-        # can write from selection == Universe (Issue 49)
-        coordinates = atoms.positions
+        # if frame is not None:
+        #     u.trajectory[frame]  # advance to frame
+        # else:
+        #     frame = u.trajectory.ts.frame
+
+        try:
+
+            # make sure to use atoms (Issue 46)
+            atoms = obj.atoms
+            # can write from selection == Universe (Issue 49)
+
+        except AttributeError:
+            if isinstance(obj, base.Timestep):
+                selection = obj.copy()
+            else:
+                raise TypeError("No Timestep found in obj argument")
+        finally:
```

Comment:
I think you can just get rid of this entire `finally` block.  `obj` is either - `AtomGroup`/`Universe` covered by `obj.atoms`, or `Timestep` which is covered by the `isinstance` call, else `TypeError`